# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../directfb/types-fix.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---


Not sure if these __{u,s}{8,16,32,64} defines should be disabled unconditionally.
At least on x86_64, this compiles however.
	--jsaw

--- DirectFB-0.9.24/include/dfb_types.h.orig	2005-07-26 22:58:50.000000000 +0200
+++ DirectFB-0.9.24/include/dfb_types.h	2006-02-13 22:03:19.000000000 +0100
@@ -2,6 +2,9 @@
 #define __DFB_TYPES_H__
 
 #include <sys/types.h>
+#include <asm/types.h>
+
+#ifdef __ASSEMBLY__
 
 #ifndef USE_KOS
 
@@ -15,6 +18,7 @@
 #define __s32 int32_t
 #define __s64 int64_t
 
+
 #else
 
 #define __u8  uint8
@@ -30,3 +34,5 @@
 #endif
 
 #endif
+
+#endif
